hypha/apply/users/middleware.py

```diff
@@ -13,3 +15,31 @@ def get_message(self, request, exception):
             return 'Your credentials are not recognised.'
 
         super().get_message(request, exception)
+
+
+class TwoFactorAuthenticationMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def is_path_allowed(self, path):
+        allowed_urls = [
+            'login/',
+            'logout/',
+            'account/',
+            'two_factor/',
+        ]
+        for url in allowed_urls:
+            if url in path:
+                return True
+        return False
+
+    def __call__(self, request):
+        # code to execute before the view
+        user = request.user
+        if settings.ENFORCE_TWO_FACTOR and user.is_authenticated and not user.is_verified() and not self.is_path_allowed(request.path):
+            return redirect('/account/two_factor/required/')
```

Comment:
Let's generate the redirection url based on the name of the url using reverse.

Reply:
Actually, I tried that first but it was giving some errors.

Reply:
I played around it, Looks like we can use name of url only after the [get_response called](https://github.com/HyphaApp/hypha/pull/2804/files#diff-1ea73a7302b3be891ad19134ba6aab169bb73c38a05c7c5c3967e9fd87a865a4R42) that means after view called in middleware. Now sure why it is like that.